### DISCUSSION TITLE
IF97 backend

### DISCUSSION THREAD

**Discussion by eliesaikali** [CoolProp/CoolProp#2237] [Q&A]
Hello,

I am trying to use the IF97 backend since I heard that it is the fastest in coolprop
It is clear from http://www.coolprop.org/fluid_properties/IF97.html that there are some limitations for this backend.

I implemented the derivatives in my code (as they are not yet available), but I am having issues with the cpmass() and viscosity() and conductivity() calls.

In practice, the method calc_Flash (in this backend) throws some exceptions in some cases of iphase_twophase.

As I am interested in cp, mu and lambda, any possible solution to fix this ?

Thanks

**Comment by ibell** [comment]
Can you please provide more information and example code for the failures?
Hard to say otherwise.

**Reply by eliesaikali** [reply]
Thanks for the reply,
Here a simple test (I converted to python)

from CoolProp.CoolProp import PropsSI

print(PropsSI('C','H',1e+06,'P',1.5e+07,'IF97::Water'))
print(PropsSI('V','H',1e+06,'P',1.5e+07,'IF97::Water'))
print(PropsSI('C','H',2e+06,'P',1.5e+07,'IF97::Water'))
print(PropsSI('V','H',2e+06,'P',1.5e+07,'IF97::Water'))

Thanks !

**Comment by henningjp** [comment]
@eliesaikali - Your first set of inputs is in the liquid phase (H < HsatL @ P=15MPa), while your second set is indeed in the 2-phase region (HsatL < H < HsatV @ P = 15MPa).  Transport properties (viscosity or thermal conductivity) and Cp/Cv are not physically meaningful in the 2-phase region and so are not defined; but are defined in the saturated liquid and saturated vapor states.  See #1973.

**Reply by eliesaikali** [reply]
Hello and thanks for your reply !

But I would like to know why you said that Cp/Cv are not physically meaningful in the 2-phase region as REFPROP & HEOS models (for example) do give values for such case ? 
Thanks

**Comment by ibell** [comment]
@henningjp is right; c_p and c_v are undefined for 0 < Q < 1. As are transport properties. There is a two-phase c_v possibility, but that is something a bit different to the normal definition.
